chitchat/src/state.rs

```diff
@@ -402,20 +403,77 @@ impl ClusterState {
     }
 }
 
+/// Score used to decide which member should be gossiped first.
+///
 /// Number of stale key-value pairs carried by the node. A key-value is considered stale if its
 /// local version is higher than the max version of the digest, also called "floor version".
-type Staleness = usize;
+#[derive(Clone, Copy, Eq, PartialEq)]
+struct Staleness {
+    is_unknown: bool,
+    num_stale_records: usize,
+}
+
+/// The ord should be considered a "priority". The higher the faster a node's
+/// information is gossiped.
+impl Ord for Staleness {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // Nodes get gossiped in priority.
+        // Unknown nodes get gossiped first.
+        self.is_unknown.cmp(&other.is_unknown).then_with(|| {
+            // Then nodes with the highest number of stale records get higher priority.
+            self.num_stale_records.cmp(&other.num_stale_records)
+        })
+    }
+}
+
+impl PartialOrd for Staleness {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
 
 /// Sorts the stale nodes in decreasing order of staleness.
-#[derive(Debug, Default)]
+#[derive(Default)]
 struct SortedStaleNodes<'a> {
     stale_nodes: BTreeMap<Staleness, Vec<StaleNode<'a>>>,
 }
 
+/// The `staleness_score` is used to decide which node should be gossiped first.
+/// `floor_version` is the version (transmitted in the digest), below which
+/// all of the records have already been received.
+///
+/// There is no such thing as a KV for version 0. So if `floor_version == 0`
+/// it means the node is entirely new.
+/// We artificially prioritize those nodes to make sure their membership (in quickwit the service
+/// key for instnace) and initial KVs spread rapidly.
+///
+/// If no KV is stale, there is nothing to gossip, and we simply return `None`:
+/// the node is not a candidate for gossip.
+fn staleness_score(node_state: &NodeState, floor_version: u64) -> Option<Staleness> {
+    let is_unknown = floor_version == 0u64;
+    let num_stale_kv = if is_unknown {
+        node_state.num_key_values()
+    } else {
+        node_state.stale_key_values(floor_version).count()
+    };
+    // We don't return None if the node is unknown yet because we want to make
```

Comment:
the comment is not very clear. The function comment is clearer so I would remove this one